akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -1181,6 +1181,7 @@ private[akka] class Shard(
 
   override def postStop(): Unit = {
     passivateIntervalTask.foreach(_.cancel())
+    lease.foreach(_.release())
```

Comment:
do we need something similar in ClusterSingleton or can that transition be trusted to be run on postStop?

Reply:
It releases in two of the state transitions (from Oldest to any other state, from AcquiringLease to anything but Oldest), I just checked that there is something there, did not investigate if there is anything incorrect with it.

Reply:
I'm not sure if it will pass through these transitions for example in case of ActorSystem termination. Maybe there is already something else that would release the lease in case of ActorSystem termination?

Reply:
so we should look some more at this before merging?

Reply:
I think we can merge this and investigate releasing on system termination as a separate task.

Reply:
ok, please create an issue

Reply:
https://github.com/akka/akka/issues/31737